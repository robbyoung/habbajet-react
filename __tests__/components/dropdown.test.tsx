import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from '../../app/components/dropdown';

describe('Dropdown Component', () => {
    it('can render a dropdown', () => {
        const component = renderer.create(
            <Dropdown
                title="Test Dropdown"
                selected={'one'}
                options={[
                    {key: 'Option one', value: 'one'},
                    {key: 'Option two', value: 'two'},
                ]}
                onValueChange={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will default to "None Selected"', () => {
        const component = renderer.create(
            <Dropdown
                title="Test Dropdown"
                options={[
                    {key: 'Option one', value: 'one'},
                    {key: 'Option two', value: 'two'},
                ]}
                onValueChange={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
