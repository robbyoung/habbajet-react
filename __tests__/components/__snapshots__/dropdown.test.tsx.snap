// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component can render a dropdown 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "color": "#959595",
        "fontFamily": "Abel",
        "fontSize": 20,
      }
    }
  >
    Test Dropdown
  </Text>
  <View
    style={
      Object {
        "borderColor": "#959595",
        "borderRadius": 5,
        "borderWidth": 1,
      }
    }
  >
    <View>
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Option one",
              "textColor": undefined,
              "value": "one",
            },
            Object {
              "label": "Option two",
              "textColor": undefined,
              "value": "two",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            Object {
              "backgroundColor": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Dropdown Component will default to "None Selected" 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "color": "#959595",
        "fontFamily": "Abel",
        "fontSize": 20,
      }
    }
  >
    Test Dropdown
  </Text>
  <View
    style={
      Object {
        "borderColor": "#959595",
        "borderRadius": 5,
        "borderWidth": 1,
      }
    }
  >
    <View>
      <RNCPicker
        items={
          Array [
            Object {
              "label": "None Selected",
              "textColor": undefined,
              "value": "",
            },
            Object {
              "label": "Option one",
              "textColor": undefined,
              "value": "one",
            },
            Object {
              "label": "Option two",
              "textColor": undefined,
              "value": "two",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            Object {
              "backgroundColor": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
